src/eval/RunnerCLI.ml

```diff
@@ -144,14 +144,16 @@ let validate_main usage =
     if
       String.(
         !f_input_message <> ""
-        && (!f_input_state <> ""
-            && (!i_ipc_address <> "" || Option.is_some !v_balance)
-           || !f_input_state = ""
-              && (!i_ipc_address = "" || Option.is_none !v_balance)))
+        && ((!f_input_state <> ""
+             && (!i_ipc_address <> "" || Option.is_some !v_balance)
+            || !f_input_state = ""
+               && (!i_ipc_address = "" || Option.is_none !v_balance))
+           || not (Bool.equal (!f_input_blockchain = "") (!f_input_state = ""))
+           ))
     then
       msg
-      ^ "Input message provided, but either none or both of input state / (IPC \
-         address and balance) provided\n"
+      ^ "Input message provided, but either none or both of input state, \
+         blockchain info / (IPC address and balance) provided\n"
```

Comment:
This actually suggests that for non-ipc purposes we ought to embed `BLOCKNUMBER` in the state json... 

Probably better to fix that in a different PR, but still.

Reply:
> 

I agree. Maybe `BLOCKNUMBER` must now be part of state JSON, but it's too large a change (mostly mechanical, but still large in terms of lines of code) to push here.

Reply:
Agreed.